ethapi/block_cert_json_test.go

```diff
@@ -90,3 +92,93 @@ func TestBlockCertificateToJson(t *testing.T) {
 	require.Equal(bitset, json.Signers)
 	require.Equal(sig, json.Signature)
 }
+
+func TestBlockCertificate_JsonEncodingMatchesExpectedFormat(t *testing.T) {
+	tests := map[string]cert.BlockCertificate{
+		"empty":     makeEmptyBlockCert(),
+		"non-empty": makeTestBlockCert(t),
+	}
+
+	for name, cert := range tests {
+		t.Run(name, func(t *testing.T) {
+			validateBlockCertJsonFormat(t, cert)
+		})
+	}
+}
+
+func validateBlockCertJsonFormat(t *testing.T, cert cert.BlockCertificate) {
+	hashRegex := `"0x[0-9a-f]{64}"`
+	tests := map[string]string{
+		"chainId":   `"chainId":\d+`,
+		"number":    `"number":\d+`,
+		"hash":      `"hash":` + hashRegex,
+		"stateRoot": `"stateRoot":` + hashRegex,
+		"signers":   `"signers":("0x[0-9a-f]+"|null)`,
+		"signature": `"signature":"0x[0-9a-f]{192}"`,
+	}
+
+	for name, regex := range tests {
+		t.Run(name, func(t *testing.T) {
+			require := require.New(t)
+			data, err := json.Marshal(toJsonBlockCertificate(cert))
+			require.NoError(err)
+			require.Regexp(regexp.MustCompile(regex), string(data))
+		})
+	}
+}
+
+func TestBlockCertificate_ContainsExpectedValues_EmptyCertificate(t *testing.T) {
+	require := require.New(t)
+	emptyCert := makeEmptyBlockCert()
+	data, err := json.Marshal(toJsonBlockCertificate(emptyCert))
+	require.NoError(err)
+	require.Contains(string(data), `"chainId":0`)
+	require.Contains(string(data), `"number":0`)
+	require.Contains(string(data), `"hash":"0x`)
+	require.Contains(string(data), `"stateRoot":"0x`)
+	require.Contains(string(data), `"signers":null`)
+	require.Contains(string(data), `"signature":"0x`)
+}
+
+func TestBlockCertificate_ContainsExpectedValues_NonEmptyCertificate(t *testing.T) {
+	require := require.New(t)
+	testCert := makeTestBlockCert(t)
+	agg := testCert.Signature()
+	signers, err := json.Marshal(agg.Signers())
+	require.NoError(err)
+
+	data, err := json.Marshal(toJsonBlockCertificate(testCert))
+	require.NoError(err)
+
+	require.Contains(string(data), `"chainId":123`)
+	require.Contains(string(data), `"number":456`)
+	require.Contains(string(data), fmt.Sprintf(`"hash":"%v"`, common.Hash{0x1}))
+	require.Contains(string(data), fmt.Sprintf(`"stateRoot":"%v"`, common.Hash{0x2}))
+	require.Contains(string(data), fmt.Sprintf(`"signers":%v`, string(signers)))
+	require.Contains(string(data), fmt.Sprintf(`"signature":"%v"`, agg.Signature()))
+}
+
+func makeEmptyBlockCert() cert.BlockCertificate {
+	return cert.NewCertificateWithSignature(
+		cert.NewBlockStatement(0, 0, common.Hash{}, common.Hash{}),
+		cert.NewAggregatedSignature[cert.BlockStatement](
+			cert.BitSet[scc.MemberId]{}, bls.Signature{}),
+	)
```

Comment:
This should be the same as
```suggestion
	return cert.BlockCertificate{}
```
and then it is cleaner to just inline it.